src/Cache/Generator/HeaderCacheKeyGenerator.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Http\Client\Common\Plugin\Cache\Generator;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Generate a cache key by specify what headers you want to vary on.
```

Comment:
i'd like to use a different word than "vary" here to avoid confusion with the `Vary` response header.